### PR TITLE
feature: expose list of banned users to chat endpoint

### PR DESCRIPTION
**Pull request by webatspeed**

### CODE REVIEW

File: src/main/java/de/caritas/cob/userservice/api/adapters/rocketchat/dto/Room.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+@Data
+public class Room {
+
+  @JsonProperty("_id")
+  private String id;
+
+  private String name;
+
+  private List<String> muted;
```
Comment (hill-daniel) [inline src/main/java/de/caritas/cob/userservice/api/adapters/rocketchat/dto/Room.java:15]: Does the list contain ids of users which are muted?
I suggest to rename it to mutedUserIds.

File: src/main/java/de/caritas/cob/userservice/api/adapters/web/controller/UserDtoMapper.java [hunk 2]
```diff
@@ -53,4 +54,9 @@ public TwoFactorAuthDTO twoFactorAuthDtoOf(OtpInfoDTO otpInfoDTO, Boolean encour
         "email", email
     );
   }
+
+  @SuppressWarnings("unchecked")
+  public List<String> bannedChatUserIdsOf(Map<String, Object> chatMetaInfoMap) {
+    return (List<String>) chatMetaInfoMap.get("mutedUsers");
```
Comment (hill-daniel) [inline src/main/java/de/caritas/cob/userservice/api/adapters/web/controller/UserDtoMapper.java:60]: Muted and Banned are two different concepts for me.
Are they just muted or banned from accessing / reading the chat?

File: src/main/java/de/caritas/cob/userservice/api/adapters/rocketchat/dto/Room.java (hunk 1 above)
Comment (webatspeed) [reply src/main/java/de/caritas/cob/userservice/api/adapters/rocketchat/dto/Room.java:15]: Yes, IDs. This is how the response by the Rocket.Chat API looks like. Thus, I cannot rename only the field only unless I add a JsonProperty annotation for the special treatment. Would you like that or should we leave it like that?

Comment (hill-daniel) [reply src/main/java/de/caritas/cob/userservice/api/adapters/rocketchat/dto/Room.java:15]: I see. Leave it then 👍

File: src/main/java/de/caritas/cob/userservice/api/adapters/web/controller/UserDtoMapper.java (hunk 2 above)
Comment (webatspeed) [reply src/main/java/de/caritas/cob/userservice/api/adapters/web/controller/UserDtoMapper.java:60]: That's right. This is mapping, so two (lingo) worlds collide. Rocket.Chat has two similar features: kick and mute. The former is not permanently, the latter still displays the messages of the chat. "Ban" is permanently (until the end of the current chat instance) and removes the messages from the display (done by the frontend.)

**Review by hill-daniel — APPROVED** [review]

**Pull request merged**